src/OpenTelemetry/Instrumentation/PropertyFetcher.cs

```diff
@@ -45,6 +45,11 @@ public PropertyFetcher(string propertyName)
         /// <returns>Property fetched.</returns>
         public T Fetch(object obj)
         {
+            if (obj == null)
+            {
+                return default;
```

Comment:
I feel like this should be a throw. Caller should check if the object is null before calling. PropertyFetcher is essentially analogous to `obj.Property` which would throw if obj is null. Returning default could be surprising and requires a post-check. @reyang @cijothomas Any thoughts?

Reply:
i was in doubt when i did that as well. the only problem that each time we will have to verify the object before use.

Reply:
How about... `Fetch` throws if obj is null and we add a `bool TryFetch(object obj, out T value)` for this case?

Reply:
Can we have that return false if the wrong object type (not the first one) is passed into it?

Reply:
@eksmith, what is your idea? What would be the difference from TryFetch?

Reply:
Following @CodeBlanch  idea, i was planning to do this:
```csharp
public bool TryFetch(object obj, out T value)
        {
            if (obj == null)
            {
                value = default;
                return false;
            }

            value = Fetch(obj);
            return true;
        }
```

Reply:
The `TypedPropertyFetch` gets passed the wrong wrong incompatible object type and just returns default. That behavior is super misleading and caused me great confusion. So I'm saying that if you add a `TryFetch` that returns a bool, which I think is great, then I'd like it to return false meaning it was unable to get the value.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/f14838c8de5c78fb29051bcf7990ded516304b1f/src/OpenTelemetry/Instrumentation/PropertyFetcher.cs#L106

Reply:
i can check in that if the value is default, if yes, so we couldnt get the value....but that would be an assumption...what do u think?

Reply:
Yes, that is a very bad assumption. I would think we would change the `TypedPropertyFetch` to use the `TryGet` pattern so we know exactly what happened.

Reply:
@ejsmith, not sure how that works exactly. at least the property doesnt contain the trygetmethod. would like to take a look? or should we do 2 parts: 1 with the `tryfetch` and other propagating with `tryget`?